### ISSUE TITLE
[Bug]: int = float is evaluated as true

### ISSUE THREAD

**Issue by SilvaXiang** [matrixorigin/matrixone#1699]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.1.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

1.create table
`create table t (a int);`
2.insert data
`insert into t2 values (1),(2),(3);`
3.query
`select * from t2 where a=1.2;`
4.get the wrong result
1

### Expected Behavior

empty set

### Steps to Reproduce

```markdown
1.create table
`create table t (a int);`
2.insert data
`insert into t2 values (1),(2),(3);`
3.query
`select * from t2 where a=1.2;`
`select 1=1.2,a from t2;`
```

**Comment by m-schen** [comment]
this bug is due to we do some type-conversion incorrectly.
and we are going to solving this bug and other type-conversion bugs after we have decided our type conversion rule.

**Comment by m-schen** [comment]
these bug is same to #1690

**Comment by broccoliSpicy** [comment]
The current prune function has some logical errors(prune.go file in pkg/sql/plan), for your test case, we incorrectly cast the 1.2 to the same type as a(int type), we fixed this issue in PR#1689. However, we want to have a deterministic type system for operations between different types, we are currently surveying the ISO SQL standard and other database systems for this and therefore this PR#1689 is still pending.

**Comment by broccoliSpicy** [comment]
In our current implementation, the file pkg/colexec/extend/overload/init.go maintains a cast rule map(OperatorCastRule) for operations between different types and we cast operations between floats type and integer type to both float64 type before operation.
